test/common/ops/buy/buy.js

```diff
@@ -197,4 +201,62 @@ describe('shared.ops.buy', () => {
       done();
     }
   });
+
+  it('Buy mount (hourglass)', () => {
+    user.purchased.plan.consecutive.trinkets = 1;
+
+    const type = 'mounts';
+    const key = 'MagicalBee-Base';
+
+    buy(user, {
+      params: {
+        key,
+        type,
+      },
+      quantity: 1,
+    });
+
+    expect(user.items.mounts[key]).to.be.true;
+    expect(pinnedGearUtils.removeItemByPath.calledOnce).to.equal(true);
+  });
+
+  it('Buy pet (hourglass)', () => {
+    user.purchased.plan.consecutive.trinkets = 1;
+
+    const type = 'pets';
+    const key = 'MagicalBee-Base';
+
+    buy(user, {
+      params: {
+        key,
+        type,
+      },
+      quantity: 1,
+    });
+
+    expect(user.items.pets[key]).to.equal(5);
+    expect(pinnedGearUtils.removeItemByPath.calledOnce).to.equal(true);
+  });
+
+  it('Buy background (hourglass)', () => {
+    user.purchased.plan.consecutive.trinkets = 1;
+
+    const type = 'backgrounds';
+    const key = 'airship';
+
+    buy(user, {
+      params: {
+        key,
+        type,
+      },
+      quantity: 1,
+    },
+    analytics,
+    {
+      hourglass: true,
+    });
+
+    expect(user.purchased.background[key]).to.be.true;
+    expect(pinnedGearUtils.removeItemByPath.calledOnce).to.equal(true);
```

Comment:
I think it would be the better - to let the real function be used and then check on the pinnedItems after that

Reply:
agreed!

Reply:
Sorry, I don't quite understand.  I am calling "buy".  What do you mean by "the real function"?

Reply:
I believe what's meant is, rather than spying on the `removeItemByPath` function and expecting it to be called once, we should put the item into the test user's `pinnedItems` array, run the `buy` function, and then `expect` the pinned item to be missing at the ent.